elastic/kibana
# [Search Session] Fix integration in vega, timelion and TSVB

## Summary

This pr address some search session integration issues in vega, timelion, and TSVB panels. 

Part of https://github.com/elastic/kibana/issues/61738
Follow up on https://github.com/elastic/kibana/pull/82835, https://github.com/elastic/kibana/pull/82232, https://github.com/elastic/kibana/pull/82229/

* Similar to visualize/lens integration pass local `sessionId` received from expression pipeline instead of getting it from global data service 
* Because timelion and TSVB don't use client-side data service:
  * Pass `isRestore` and `isStored` boolean flags to the server
  * Call `trackSearch/untrackSearch` on a session

### How to test
In `kibana.yml`: 
```
xpack.data_enhanced.search.sendToBackground.enabled: true
```

1. Create a dashboard with vega, timelion, and TSVB panels. Use absolute time range.
2. Save the session.
3. Get a session id from an inspector or network request. Append it to your dashboard URL: `dashboardUrl&searchSessionId={your-id}`
4. Use this URL, all panels should be loaded and the session indicator should be in "restored" state.

to make sure we are actually restoring data for all the panels try augmenting the dashboard state in the URL, e.g. changing time, or use a fake session id in the URL. All panels should fail.

### For maintainers
